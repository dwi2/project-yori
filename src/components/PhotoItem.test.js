import React from 'react'
import PhotoItem from '../../src/components/PhotoItem';
import renderer from 'react-test-renderer';

test('Render meta of photo of extra long member name', () => {
  const photo = {
    photo_group: '乃木坂46',
    photo_member: '吉田綾乃クリスティー',
    photo_costume: '日常',
    photo_type: 'ヨリ',
    photo_number: 20,
    photo_folder: '乃木坂',
    photo_tag: []
  };

  const component = renderer.create(
    <PhotoItem photo={photo} key={`photo-item-${0}`}/>,
  );
  const instance = component.root;
  expect(instance.findByProps({className: 'photo-name extra-small-text'})).toBeTruthy();

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
