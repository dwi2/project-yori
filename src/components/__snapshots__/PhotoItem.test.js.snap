// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render meta of photo of extra long member name 1`] = `
<div
  className="photo-item-wrapper"
>
  <div
    className="photo-image"
  >
    吉
  </div>
  <div
    className="photo-data"
  >
    <div
      className="photo-data-meta"
    >
      <h2
        className="photo-name extra-small-text"
      >
        吉田綾乃クリスティー
      </h2>
      <h2
        className="photo-type"
      >
        ヨリ
      </h2>
      <h3
        className="photo-costume"
      >
        日常
      </h3>
    </div>
    <div
      className="photo-data-number"
    >
      <h3>
        20
      </h3>
    </div>
  </div>
</div>
`;
